ReactiveCocoa/Swift/Lifetime.swift

```diff
@@ -3,9 +3,54 @@ import enum Result.NoError
 /// Represents the lifetime of an object, and provides a hook to observe when
 /// the object deinitializes.
 public final class Lifetime {
-	/// A signal that sends a Completed event when the lifetime ends.
+	/// MARK: Type properties and methods
+
+	/// A `Lifetime` that has already ended.
+	public static var empty: Lifetime {
+		return Lifetime(ended: .empty)
+	}
+
+	/// MARK: Instance properties
+
+	/// A signal that sends a `completed` event when the lifetime ends.
 	public let ended: Signal<(), NoError>
 
+	/// MARK: Initializers
+
+	/// Initialize a `Lifetime` object with the supplied ended signal.
+	///
+	/// - parameters:
+	///   - signal: The ended signal.
+	private init(ended signal: Signal<(), NoError>) {
+		ended = signal
+	}
+
+	/// Initialize a `Lifetime` object which ends when the supplied producer
+	/// terminates.
+	///
+	/// - parameters:
+	///   - producer: The producer to be observed for termination.
+	public convenience init<U, E: Error>(_ producer: SignalProducer<U, E>) {
```

Comment:
:-1:

This initialization has side effects. Additionally, I don't see how this would be useful given that each subscriber sees potentially a different lifetime...